File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/cpp/au_350_c_bor_i64_bool.cpp
// arch-unit 350  --  c  `a | b`  lhs=int64_t rhs=bool
// symbol op_365   outcome LIFTED   2 arch-opcodes
//
// the arch-unit, arch-opcode by arch-opcode:
//   c.or a0, a1                          integer    operator:|
//   c.jr ra                              integer    return
//
// answer: int64_t, 64 bits.  parameters are operand bit patterns.
#include "arch_units_int.hpp"

namespace archunits {

uint64_t au_350_c_bor_i64_bool(uint64_t p0, uint64_t p1)
{
    /* a0: operand `a` (int64_t) arrives in a0 */
    const uint64_t v1 = p0;
    /* a1: operand `b` (bool) zero-extended to XLEN */
    const uint64_t v2 = ((p1) & UINT64_C(0x1));
    const uint64_t v3 = au_or(v1, v2);
    /* the answer is int64_t, 64 bits */
    return v3;
}

}  // namespace archunits

File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/java/au_350_c_bor_i64_bool.java
// arch-unit 350  --  c  `a | b`  lhs=int64_t rhs=bool
// symbol op_365   outcome LIFTED   2 arch-opcodes
//
// the arch-unit, arch-opcode by arch-opcode:
//   c.or a0, a1                          integer    operator:|
//   c.jr ra                              integer    return
//
// answer: int64_t, 64 bits.  parameters are operand bit patterns.
public final class au_350_c_bor_i64_bool {

    public static long au_350_c_bor_i64_bool(long p0, long p1) {
        // a0: operand `a` (int64_t) arrives in a0
        final long v1 = p0;
        // a1: operand `b` (bool) zero-extended to XLEN
        final long v2 = ((p1) & 0x1L);
        final long v3 = AuInt.au_or(v1, v2);
        // the answer is int64_t, 64 bits
        return v3;
    }
}
